### DISCUSSION TITLE
How translations work in storefront?

### DISCUSSION THREAD

**Discussion by demodes** [vendurehq/vendure#1544] [Q&A]
Hi, 
I don't know where should I ask this. I wasn't able to find out how to change languages in the storefront. I tried the official demo. I add german text in the admin dashboard for a laptop product, but the text in the storefront did not change. I am probably doing something wrong. I just don't know what. Can you help me with this, please?
![vendure](https://user-images.githubusercontent.com/23063636/166230245-9d5232d6-6c76-435a-83fd-b5b6f6e5acb3.png)

**Comment by michaelbromley** [comment]
Hi! This is covered [here](https://www.vendure.io/docs/storefront/shop-api-guide/#universal-parameters) - in short, add a query parameter `?languageCode=de` to your API call.

**Reply by demodes** [reply]
Thanks. When I add it to the API call, will it also change the functional texts in storefront? I mean "add to cart", "buy", "checkout", etc.

**Reply by michaelbromley** [reply]
No, the storefront ui text is completely out of the scope of Vendure. You would have to find a i18n solution compatible with your framework of choice.

**Comment by demodes** [comment]
Hi, thank you for the explanation.
Is it possible to create different storefronts (each with different language) and connect them to different channels? For example: Can I hardcode some storefront in English language and connect it to eng channel and another storefront in German language and connect it to the German channel. Does Vendure support different storefronts connected to the same backend?

**Reply by michaelbromley** [reply]
Yes, certainly! You'd just need to set the `vendure-token` (https://www.vendure.io/docs/storefront/shop-api-guide/#universal-parameters) variable to specify which Channel you want that storefront to interact with.
